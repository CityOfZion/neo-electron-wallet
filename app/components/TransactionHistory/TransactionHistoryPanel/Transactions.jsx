// @flow
import React from 'react'
import classNames from 'classnames'

import Transaction from '../../Blockchain/Transaction'
import TransactionList from '../../Blockchain/Transaction/TransactionList'
import LogoWithStrikethrough from '../../LogoWithStrikethrough'

import styles from './Transactions.scss'

type Props = {
  className?: string,
  transactions: Array<Object>,
}

export default class Transactions extends React.Component<Props> {
  render() {
    const { className, transactions } = this.props

    if (transactions.length === 0) {
      return (
        <div className={classNames(styles.noTransactions, className)}>
          <LogoWithStrikethrough />
        </div>
      )
    }

    return (
      <TransactionList className={className} alternateRows>
        {transactions.map((tx, i) => <Transaction tx={tx} key={i} />)}
      </TransactionList>
    )
  }
}
