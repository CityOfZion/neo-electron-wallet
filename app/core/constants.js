// @flow
export const NETWORK = {
  MAIN: 'MainNet',
  TEST: 'TestNet'
}

export const EXPLORER = {
  NEO_TRACKER: 'Neotracker',
  ANT_CHAIN: 'Antchain'
}

export const ASSETS = {
  NEO: 'NEO',
  GAS: 'GAS'
}

export const ASSETS_LABELS = {
  [ASSETS.NEO]: 'Neo',
  [ASSETS.GAS]: 'Gas'
}

export const BIP44_PATH =
  '8000002C' +
  '80000378' +
  '80000000' +
  '00000000' +
  '00000000'
