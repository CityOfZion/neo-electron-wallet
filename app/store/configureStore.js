import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';
// export type GlobalState = {}
export var INITIAL_STATE = {};
var loggerMiddleware = createLogger();
function configureStore(initialState) {
    if (initialState === void 0) { initialState = INITIAL_STATE; }
    return createStore(rootReducer, initialState, process.env.NODE_ENV === 'production'
        ? applyMiddleware(thunk)
        : applyMiddleware(thunk, loggerMiddleware));
}
export var store = configureStore();
