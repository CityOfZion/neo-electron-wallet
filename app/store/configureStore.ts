import { createStore, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'
import rootReducer from '../reducers/rootReducer'

// export type GlobalState = {}
export const INITIAL_STATE = {}

const loggerMiddleware = createLogger()

function configureStore(initialState = INITIAL_STATE) {
  return createStore(
    rootReducer,
    initialState,
    process.env.NODE_ENV === 'production'
      ? applyMiddleware(thunk)
      : applyMiddleware(thunk, loggerMiddleware),
  )
}
export const store = configureStore()
