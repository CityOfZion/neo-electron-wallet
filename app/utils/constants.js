const explorers = [
  {
    id: 'neotracker',
    name: 'NEO Tracker',
    main: 'https://neotracker.io',
    test: 'https://testnet.neotracker.io',
    getTransactionLink(net, txid) {
      const base = net === 'MainNet' ? this.main : this.test;
      return base + '/tx/' + txid;
    },
  },
  {
    id: 'antchain',
    name: 'Antchain',
    main: 'http://antcha.in',
    test: 'http://testnet.antcha.in',
    getTransactionLink(net, txid) {
      const base = net === 'MainNet' ? this.main : this.test;
      return base + '/tx/hash/' + txid;
    },
  },
];
const idToExplorer = {};
explorers.forEach(explorer => {
  idToExplorer[explorer.id] = explorer;
});

export const BLOCKCHAIN_EXPLORERS = idToExplorer;
