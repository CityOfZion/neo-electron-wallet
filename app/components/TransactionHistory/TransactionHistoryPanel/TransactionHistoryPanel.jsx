// @flow
import React from 'react'
import classNames from 'classnames'

import Transactions from './Transactions'
import Panel from '../../Panel'

import styles from './TransactionHistoryPanel.scss'

type Props = {
  className: ?string,
  transactions: Array<Object>
}

export default class TransactionHistory extends React.Component<Props> {
  render () {
    const { className, transactions } = this.props

    return (
      <Panel className={classNames(styles.transactionHistoryPanel, className)} renderHeader={this.renderHeader}>
        <Transactions className={styles.transactions} transactions={transactions} />
      </Panel>
    )
  }

  renderHeader = () => {
    return <span>Transaction History</span>
  }
}
