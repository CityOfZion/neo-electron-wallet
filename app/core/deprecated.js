// @flow
import { get } from 'lodash'

import { ID as NETWORK_ID } from '../actions/networkActions'
import { getNetworks } from '../core/networks'

const PREFIX = 'api'

// TODO: Module functions like `doClaimNotify` should be refactored into function props exposed by
//       higher-order components.  Data normally provided by functions like `getNetwork` can instead
//       be retrieved via the `withData` higher-order component.
export const getNetworkId = (state: Object) => {
  return get(state, `${PREFIX}.${NETWORK_ID}.data`)
}

export const getNetwork = (state: Object) => {
  const selectedNetworkId = getNetworkId(state)
  const networks = getNetworks()
  const networkItem = networks.find(({ id, value }) => id === selectedNetworkId) || networks[0]

  return networkItem.network
}

export const getNetworkById = (networkId: string) => {
  const networks = getNetworks()
  const networkItem = networks.find(({ id, value }) => id === networkId) || networks[0]
  return networkItem.network
}
