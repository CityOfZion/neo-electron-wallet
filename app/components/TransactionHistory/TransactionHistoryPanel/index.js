// @flow
import { compose } from 'recompose'

import TransactionHistoryPanel from './TransactionHistoryPanel'
import transactionHistoryActions from '../../../actions/transactionHistoryActions'
import withData from '../../../hocs/api/withData'
import withProgressComponents from '../../../hocs/api/withProgressComponents'
import withNetworkData from '../../../hocs/withNetworkData'
import withAuthData from '../../../hocs/withAuthData'
import alreadyLoaded from '../../../hocs/api/progressStrategies/alreadyLoadedStrategy'
import Loading from '../../../containers/App/Loading'
import Failed from '../../../containers/App/Failed'
import { LOADING, FAILED } from '../../../values/state'

const mapTransactionsDataToProps = (transactions) => ({
  transactions
})

export default compose(
  withNetworkData(),
  withAuthData(),
  withData(transactionHistoryActions, mapTransactionsDataToProps),
  withProgressComponents(transactionHistoryActions, {
    [LOADING]: Loading,
    [FAILED]: Failed
  }, {
    strategy: alreadyLoaded
  })
)(TransactionHistoryPanel)
