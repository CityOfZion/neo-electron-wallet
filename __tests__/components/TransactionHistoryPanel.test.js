import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import { shallow, mount } from 'enzyme'
import { merge } from 'lodash'

import TransactionHistoryPanel from '../../app/components/TransactionHistory/TransactionHistoryPanel'
import { MAIN_NETWORK_ID, EXPLORERS } from '../../app/core/constants'
import { LOADED } from '../../app/values/state'

const initialState = {
  api: {
    AUTH: {
      address: 'AWy7RNBVr9vDadRMK9p7i7Z1tL7GrLAxoh',
      wif: 'L4SLRcPgqNMAMwM3nFSxnh36f1v5omjPg3Ewy1tg2PnEon8AcHou'
    },
    NETWORK: {
      batch: false,
      state: LOADED,
      loadedCount: 1,
      data: MAIN_NETWORK_ID
    },
    SETTINGS: {
      batch: false,
      state: LOADED,
      loadedCount: 1,
      data: {
        blockExplorer: EXPLORERS.NEO_TRACKER
      }
    },
    TRANSACTION_HISTORY: {
      batch: false,
      state: LOADED,
      loadedCount: 1,
      data: []
    }
  }
}

const transactions = [
  {
    NEO: '50',
    GAS: '0.00000000',
    txid: '76938979'
  },
  {
    NEO: '0',
    GAS: '0.40000000',
    txid: '76938980'
  }
]

const setup = (state = initialState, shallowRender = true) => {
  const store = configureStore([thunk])(state)

  let wrapper
  if (shallowRender) {
    wrapper = shallow(<TransactionHistoryPanel store={store} />)
  } else {
    wrapper = mount(
      <Provider store={store}>
        <TransactionHistoryPanel />
      </Provider>
    )
  }

  return {
    store,
    wrapper
  }
}

describe('TransactionHistoryPanel', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('correctly renders no transaction history', () => {
    const { wrapper } = setup(initialState, false)

    const transactionList = wrapper.find('#transactionList')
    expect(transactionList.children().length).toEqual(0)
  })

  test('correctly renders with NEO and GAS transaction history', () => {
    const transactionState = merge({}, initialState, {
      api: { TRANSACTION_HISTORY: { data: transactions } }
    })
    const { wrapper } = setup(transactionState, false)

    const transactionList = wrapper.find('#transactionList')
    expect(transactionList.children().length).toEqual(2)
    expect(transactionList.childAt(0).find('.txid').first().text()).toEqual(transactions[0].txid)
    expect(transactionList.childAt(1).find('.txid').first().text()).toEqual(transactions[1].txid)
    expect(transactionList.childAt(0).find('.amountNEO').text()).toEqual('50 NEO')
    expect(transactionList.childAt(1).find('.amountGAS').text()).toEqual('0.40000000 GAS')
  })
})
