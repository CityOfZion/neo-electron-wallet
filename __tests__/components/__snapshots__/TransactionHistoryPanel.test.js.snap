// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHistoryPanel renders without crashing 1`] = `
<withProgress(withProgressComponents(Connect(withData(TransactionHistory))))
  dispatch={[Function]}
  progress="LOADED"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    t {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
